hazelcast/include/hazelcast/client/connection/ClusterListenerThread.h

```diff
@@ -56,8 +56,6 @@ namespace hazelcast {
 
                 void setThread(util::Thread *);
```

Comment:
Can we also move the initialization of this thread from ClusterService to ClusterListenerThread and remove setThread  also. And also make staticRun and run methods private.